Automattic/newspack-blocks
# Homepage Posts: Deduplicate posts in editor when there are multiple blocks

### All Submissions:






### Changes proposed in this Pull Request:

This PR changes the editor to match the de-duplication logic on the front-end.

This changes the Homepage Articles block to use a `wp.data` store to query instead of `getEntityRecords`. The store takes requests and returns a list of posts that are unique to the block. If the same post would show up in multiple blocks, the first block on the page will get it and the rest of the blocks will not.

### How to test the changes in this Pull Request:

1. Add multiple Homepage Articles blocks to the page
2. Configure them to return the same post
3. Confirm that the same post does not show up multiple times.
4. Check that in Specific Post Mode, the posts are returning correctly
